apps/server/src/infra/schulconnex-client/schulconnex-rest-client.ts

```diff
@@ -86,13 +88,17 @@ export class SchulconnexRestClient implements SchulconnexApiInterface {
 	private async requestClientCredentialToken(): Promise<OAuthTokenDto> {
 		const { tokenEndpoint, clientId, clientSecret } = this.options;
 
+		if (!this.checkOptions()) {
+			return Promise.reject(new Error('Missing configuration for SchulconnexRestClient'));
+		}
+
 		const payload: ClientCredentialsGrantTokenRequest = new ClientCredentialsGrantTokenRequest({
-			client_id: clientId,
-			client_secret: clientSecret,
+			client_id: clientId ?? '',
```

Comment:
checkOptions above makes sure this is set, but ok...

Reply:
Yes, but the ide complains... i am not happy about those checks everywhere. So i created a ticket to refactor it :)